src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.ICSharpProjectSite.cs

```diff
@@ -124,9 +123,14 @@ public int GetValidStartupClasses(IntPtr[] classNames, ref int count)
             // If classNames is NULL, then we need to populate the number of valid startup
             // classes only
             var project = Workspace.CurrentSolution.GetProject(VisualStudioProject.Id);
-            var compilation = project.GetCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
+            if (project == null)
```

Comment:
So why is this happening where we are being asked but the project isn't around? That seems very broken.

Reply:
In testing this cannot happen.